WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/history/ScanHistoryViewModel.kt

```diff
@@ -1,17 +1,44 @@
 package org.wordpress.android.ui.jetpack.scan.history
 
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
 import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel
+import org.wordpress.android.fluxc.store.ScanStore
+import org.wordpress.android.fluxc.store.ScanStore.FetchScanHistoryPayload
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.viewmodel.ScopedViewModel
 import javax.inject.Inject
+import javax.inject.Named
 
-class ScanHistoryViewModel @Inject constructor() : ViewModel() {
+class ScanHistoryViewModel @Inject constructor(
+    private val scanStore: ScanStore,
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+) : ScopedViewModel(mainDispatcher) {
     private var isStarted = false
 
+    private val _threats = MutableLiveData<List<ThreatModel>>()
+    val threats: LiveData<List<ThreatModel>> = _threats
+
     lateinit var site: SiteModel
 
     fun start(site: SiteModel) {
         if (isStarted) return
         isStarted = true
         this.site = site
+        fetchScanHistory()
+    }
+
+    private fun fetchScanHistory() {
+        launch {
+            val result = scanStore.fetchScanHistory(FetchScanHistoryPayload(site))
```

Comment:
I wasn't planning doing that, since I felt like the use case wouldn't do anything else but invoke this single method. However, I will wrap it in a use case when I add a network connection check.